src/smc_mallows_new_item_rank_updated.cpp

```diff
@@ -0,0 +1,213 @@
+//' @description Function to perform resample-move SMC algorithm where we receive a new item ranks from an existing user at each time step given an initial particle set obtained from MCMC or SMC. Each correction and augmentation is done by filling in the missing item ranks using pseudolikelihood augmentation.
+//' @param n_items Integer is the number of items in a ranking.
+//' @param R_obs 3D matrix of size n_assessors by n_items by Time containing a set of observed rankings of Time steps.
+//' @param metric A character string specifying the distance metric to use in the Bayesian Mallows Model. Available options are \code{"footrule"},
+//' \code{"spearman"}, \code{"cayley"}, \code{"hamming"}, \code{"kendall"}, and \code{"ulam"}.
+//' @param leap_size leap_size Integer specifying the step size of the leap-and-shift proposal distribution.
+//' @param N Integer specifying the number of particles.
+//' @param Time Integer specifying the number of time steps in the SMC algorithm.
+//' @param logz_estimate Estimate of the partition function, computed with \code{\link{estimate_partition_function}} in the BayesMallow R package {estimate_partition_function}.
+//' @param mcmc_kernel_app Integer value for the number of applications we apply the MCMC move kernel.
+//' @param alpha_prop_sd Numeric value of the standard deviation of the prior distribution for alpha.
+//' @param lambda Strictly positive numeric value specifying the rate parameter of the truncated exponential prior distribution of alpha.
+//' @param alpha_max  Maximum value of alpha in the truncated exponential prior distribution.
+//' @param aug_method A character string specifying the approach for filling in the missing data, options are "pseudolikelihood" or "random".
+//' @param alpha_samples_init A vector of size N by containing the initial particle set values of alpha.
+//' @param rho_samples_init 2D matrix of size N by n_items containing the initial particle set values of rho.
+//' @param aug_rankings_init 3D matrix of size n_assessors by n_items by N containing the initial particle set values of augmented rankings for R_obs.
+//' @return a 3d matrix containing: the samples of: rho, alpha and the augmented rankings, and the effective sample size at each iteration of the SMC algorithm.
+//' @param verbose Logical specifying whether to print out the progress of the
+//' SMC-Mallows algorithm. Defaults to \code{FALSE}.
+//' @param alpha_fixed Logical indicating whether to sample \code{alpha} or not.
+//' @param alpha numeric value of the scale parameter.
+//' @export
+// [[Rcpp::export]]
+Rcpp::List smc_mallows_new_item_rank_updated(
+  const unsigned int& n_items,
+  arma::cube& R_obs,
+  const std::string& metric,
+  const int& leap_size,
+  const unsigned int& N,
+  const unsigned int Time,
+  const Rcpp::Nullable<arma::vec> logz_estimate,
+  const int& mcmc_kernel_app,
+  arma::mat rho_samples_init,
+  arma::cube aug_rankings_init,
+  arma::vec alpha_samples_init = 0,
```

Comment:
Hi @anjastein,

I am currently working on the replacement of the functions. The highlighted code shows the three function arguments added by the update. To facilitate the replacement across the package (also, to preserve backwards compatibility and mitigate rewriting of tests and vignettes), it would be nice to have default values for these arguments. `alpha_samples_init` is already defaulting to 0 to make L71 work, but the others don't have a default yet. Is there something reasonable we could have in there?

Reply:
I have a suggestion for `rho_samples_init` I'm close to pushing. First we declare the object as `Rcpp::Nullable<arma::mat> rho_samples_init` and then use this at the beginning of the function:

```r
  if (rho_samples_init.isNotNull()) {
    rho_samples.slice(0) = Rcpp::as<arma::mat>(rho_samples_init);
  } else {
    rho_samples.slice(0) = initialize_rho(n_items, N).t();
  }
```

So if `rho_samples_init = NULL` on function call, we use `initialize_rho()` instead (as per the previous implementation).

Not sure what to do about `aug_rankings_init`, though, other than reimplementing all that code that was calculating the augmented rankings before the update. Hopefully there's a simpler alternative. Or maybe I could encapsulate that whole section into a new function?

Finally, as for `alpha_samples_init`, I'm thinking about creating a `initialize_alpha()` function of sorts that works in a similar way to `initialize_rho()`.

In any case, I'll take a better look at these options tomorrow. I'll push the changes as soon as I clean up my git tree a bit.

Reply:
I agree with your suggestion for `rho_samples_init` because we use the uniform distribution to sample potential values for rho for the particles.

For `alpha_sample_init`, I would suggest something that samples, using R, `rexp(1)` N times for the particles because `alpha` should be nonzero and positive. 

For `aug_rankings_init`, you could take the original partially observed dataset and use the uniform distribution to propose augmented rankings N times. I think this happens in the initialisation for `compute_mallows` and in `smc_new_users`.  I think this is what you're already suggesting when you say ` reimplementing all that code that was calculating the augmented rankings before the update`.